pandas/tests/sparse/test_frame.py

```diff
@@ -540,6 +540,37 @@ def test_setitem_array(self):
                                   self.frame['F'].reindex(index),
                                   check_names=False)
 
+    def test_setitem_spmatrix(self):
+        # GH-15634
+        tm.skip_if_no_package('scipy')
+        from scipy.sparse import csr_matrix
+
+        sdf = self.frame.copy(False)
+
+        def _equal(spm1, spm2):
+            return np.all(spm1.toarray() == spm2.toarray())
+
+        # 1d -- column
+        spm = csr_matrix(np.arange(len(sdf))).T
+        sdf['X'] = spm
+        assert _equal(sdf[['X']].to_coo(), spm)
+
```

Comment:
this comparision on the scipy side is fine, but also let's compare with assert_sparse_series/frame_equal